File: apps/admin/src/types/external-product.ts
// External product review types — matches backend AdminExternalProductResponse shape.

export type ExternalProductReviewStatus =
  | 'draft'
  | 'pending_review'
  | 'approved'
  | 'rejected'
  | 'request_changes'
  | 'hidden'

export interface ExternalProductMedia {
  id: string
  external_product_id: string
  media_type: 'image' | 'video'
  storage_key: string
  url: string
  thumbnail_url?: string
  sort_order: number
  created_at: string
}

export interface ExternalProductReviewHistory {
  id: string
  external_product_id: string
  actor_admin_id?: string
  actor_user_id?: string
  from_status?: string
  to_status: string
  reason?: string
  created_at: string
}

export interface AdminExternalProduct {
  id: string
  owner_user_id: string
  title: string
  description?: string
  external_url: string
  normalized_external_url: string
  review_status: ExternalProductReviewStatus
  rejection_reason?: string
  unsafe_url_flag: boolean
  submitted_at?: string
  approved_at?: string
  rejected_at?: string
  hidden_at?: string
  last_reviewed_by?: string
  created_at: string
  updated_at: string
  media?: ExternalProductMedia[]
  can_edit: boolean
  can_submit: boolean
  can_resubmit: boolean
  public_visible: boolean
  // Admin-only fields
  review_history?: ExternalProductReviewHistory[]
  can_approve: boolean
  can_reject: boolean
  can_hide: boolean
}

export interface AdminExternalProductListResponse {
  items: AdminExternalProduct[]
  count: number
  page: number
  limit: number
}

export const externalProductStatusLabels: Record<ExternalProductReviewStatus, string> = {
  draft: 'Draft',
  pending_review: 'Pending Review',
  approved: 'Approved',
  rejected: 'Rejected',
  request_changes: 'Changes Requested',
  hidden: 'Hidden',
}

export const externalProductStatusVariants: Record<
  ExternalProductReviewStatus,
  'default' | 'pending' | 'success' | 'error' | 'warning' | 'info'
> = {
  draft: 'default',
  pending_review: 'pending',
  approved: 'success',
  rejected: 'error',
  request_changes: 'warning',
  hidden: 'info',
}

File: apps/mobile/lib/domains/commerce/pricing/promotion/domain/entities/external_product.dart
import 'package:equatable/equatable.dart';

import 'external_product_media.dart';
import 'external_product_review_status.dart';

/// External product entity for promotion targeting.
///
/// Represents a URL-based product that goes through admin review
/// before it can be used as a promotion target.
///
/// Lifecycle: draft → pending_review → approved/rejected/hidden
class ExternalProduct extends Equatable {
  final String id;
  final String ownerUserId;
  final String title;
  final String? description;
  final String externalUrl;
  final String normalizedExternalUrl;
  final ExternalProductReviewStatus reviewStatus;
  final String? rejectionReason;
  final bool unsafeUrlFlag;
  final DateTime? submittedAt;
  final DateTime? approvedAt;
  final DateTime? rejectedAt;
  final DateTime? hiddenAt;
  final DateTime createdAt;
  final DateTime updatedAt;
  final List<ExternalProductMedia> media;
  final bool canEdit;
  final bool canSubmit;
  final bool canResubmit;
  final bool publicVisible;

  const ExternalProduct({
    required this.id,
    required this.ownerUserId,
    required this.title,
    this.description,
    required this.externalUrl,
    required this.normalizedExternalUrl,
    required this.reviewStatus,
    this.rejectionReason,
    required this.unsafeUrlFlag,
    this.submittedAt,
    this.approvedAt,
    this.rejectedAt,
    this.hiddenAt,
    required this.createdAt,
    required this.updatedAt,
    this.media = const [],
    required this.canEdit,
    required this.canSubmit,
    required this.canResubmit,
    required this.publicVisible,
  });

  /// Whether the review status shows a rejection/hidden reason
  bool get hasRejectionReason =>
      rejectionReason != null && rejectionReason!.isNotEmpty;

  /// Whether the product is approved and eligible for promotion activation
  bool get isApproved => reviewStatus == ExternalProductReviewStatus.approved;

  @override
  List<Object?> get props => [
    id,
    ownerUserId,
    title,
    description,
    externalUrl,
    normalizedExternalUrl,
    reviewStatus,
    rejectionReason,
    unsafeUrlFlag,
    submittedAt,
    approvedAt,
    rejectedAt,
    hiddenAt,
    createdAt,
    updatedAt,
    media,
    canEdit,
    canSubmit,
    canResubmit,
    publicVisible,
  ];
}
